tests/v2/validation/provisioning/rke1/provisioning_node_driver_test.go

```diff
@@ -128,119 +118,36 @@ func (r *RKE1NodeDriverProvisioningTestSuite) TestProvisioningRKE1Cluster() {
-		{"3 nodes - 1 role per node " + provisioning.StandardClientName.String(), nodeRoles2, r.standardUserClient, r.psact},
+		{"1 Node all roles " + provisioning.AdminClientName.String(), nodeRoles0, r.client},
+		{"1 Node all roles " + provisioning.StandardClientName.String(), nodeRoles0, r.standardUserClient},
+		{"2 nodes - etcd/cp roles per 1 node " + provisioning.AdminClientName.String(), nodeRoles1, r.client},
+		{"2 nodes - etcd/cp roles per 1 node " + provisioning.StandardClientName.String(), nodeRoles1, r.standardUserClient},
+		{"3 nodes - 1 role per node " + provisioning.AdminClientName.String(), nodeRoles2, r.client},
+		{"3 nodes - 1 role per node " + provisioning.StandardClientName.String(), nodeRoles2, r.standardUserClient},
 	}
-
-	var name, scaleName string
 	for _, tt := range tests {
-		subSession := r.session.NewSession()
-		defer subSession.Cleanup()
-
-		client, err := tt.client.WithSession(subSession)
-		require.NoError(r.T(), err)
-
-		for _, providerName := range r.providers {
-			provider := CreateProvider(providerName)
-			providerName := " Node Provider: " + provider.Name
-			for _, kubeVersion := range r.kubernetesVersions {
-				name = tt.name + providerName.String() + " Kubernetes version: " + kubeVersion
-				for _, cni := range r.cnis {
-					nodeTemplate, err := provider.NodeTemplateFunc(client)
-					require.NoError(r.T(), err)
-
-					name += " cni: " + cni
-					scaleName = "scaling " + name
-					r.Run(name, func() {
-						cluster, err := TestProvisioningRKE1Cluster(r.T(), client, provider, tt.nodeRoles, tt.psact, kubeVersion, cni, nodeTemplate, r.advancedOptions)
-						require.NoError(r.T(), err)
-
-						r.cluster = cluster
-					})
-
-					r.Run(scaleName, func() {
-						r.testScalingRKE1NodePools(client, provider, tt.nodeRoles, tt.psact, kubeVersion, cni, r.cluster, nodeTemplate)
-					})
-
-					r.cluster = nil
-				}
-			}
-		}
+		r.clustersConfig.NodesAndRolesRKE1 = tt.nodeRoles
+		permutations.RunTestPermutations(&r.Suite, tt.name, tt.client, r.clustersConfig, permutations.RKE1ProvisionCluster, nil, nil)
 	}
 }
 
 func (r *RKE1NodeDriverProvisioningTestSuite) TestProvisioningRKE1ClusterDynamicInput() {
-	clustersConfig := new(provisioning.Config)
-	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
-	nodesAndRoles := clustersConfig.NodesAndRolesRKE1
 
-	if len(nodesAndRoles) == 0 {
+	if len(r.clustersConfig.NodesAndRolesRKE1) == 0 {
 		r.T().Skip()
 	}
 
+	poolNames := permutations.GetRKE1TestName(r.clustersConfig.NodesAndRolesRKE1)
 	tests := []struct {
 		name   string
 		client *rancher.Client
-		psact  string
 	}{
-		{provisioning.AdminClientName.String(), r.client, r.psact},
-		{provisioning.StandardClientName.String(), r.standardUserClient, r.psact},
+		{provisioning.AdminClientName.String() + poolNames, r.client},
+		{provisioning.StandardClientName.String() + poolNames, r.standardUserClient},
```

Comment:
I still think we shouldn't add the pool names to the test name, but rather just log it. It will affect our regex a lot.